File: src/main/java/com/olacabs/checks/lint/LintIssueRegistry.java
package com.olacabs.checks.lint;

import com.android.tools.lint.client.api.IssueRegistry;
import com.android.tools.lint.detector.api.Issue;
import java.util.Collections;
import java.util.List;

/**
 * Usage in OlaChecks-v2/build.gradle. Defines all checks available in this project.
 */
@SuppressWarnings("unused")
public class LintIssueRegistry extends IssueRegistry {

  public LintIssueRegistry() {
  }

  @Override
  public List<Issue> getIssues() {
    return Collections.singletonList(RxCheckResultAnnotationEnforcer.ISSUE);
  }
}

File: src/main/java/is/uncommon/checks/LintIssueRegistry.kt
package `is`.uncommon.checks

import com.android.tools.lint.client.api.IssueRegistry

class LintIssueRegistry : IssueRegistry() {

  override val issues = listOf(
      NamedForPrimitiveTypesOfProvidersEnforcer.ISSUE,
      ImplementsSerializableDetector.ISSUE,
      RxUsedEnforcerJ.ISSUE
  )
}
